src/sidebar/services/frame-sync.ts

```diff
@@ -508,15 +517,39 @@ export class FrameSyncService {
   }
 
   /**
-   * Mark annotations as hovered.
+   * Mark annotation as hovered.
    *
    * This is used to indicate the highlights in the document that correspond
-   * to hovered annotations in the sidebar.
+   * to a hovered annotation in the sidebar.
    *
-   * @param tags - annotation $tags
+   * This function only accepts a single annotation because the user can only
+   * hover one annotation card in the sidebar at a time. Hover updates in the
+   * other direction (guest to sidebar) support multiple annotations since a
+   * user can hover multiple highlights in the document at once.
    */
-  hoverAnnotations(tags: string[]) {
+  hoverAnnotation(ann: Annotation | null) {
+    this._pendingHoverTag = null;
+
+    if (!ann) {
+      this._guestRPC.forEach(rpc => rpc.call('hoverAnnotations', []));
+      return;
+    }
```

Comment:
This guard returns before `this._store.hoverAnnotations` is invoked. Could this ever be problematic? Could it end up getting out of sync?

Perhaps I could frame this question from another angle: what's the purpose of accepting a `null` value for the method's argument?

Reply:
> Perhaps I could frame this question from another angle: what's the purpose of accepting a null value for the method's argument?

When the mouse is in the sidebar but not hovering on any annotation card, then no annotations should be hovered in any guest. Previously there was no explicit logic for this because the `hoverAnnotations` RPC call accepts a list, and so did `FrameSyncService.hoverAnnotations`, so de-hovering all annotations was done by passing an empty list to this method.